File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/09-Queue-Deque/LeetCode/LeetCode-0225-2.java
class MyStack {
    Queue<Integer> Q = new LinkedList<>();

    public MyStack() {}
    
    public void push(int x) {
        int size = Q.size();
        Q.add(x);
        while (size-- > 0){
            Q.add(Q.remove());
        } // end-for
    }
    
    public int pop() {return Q.remove();}
    
    public int top() {return Q.peek();}
    
    public boolean empty() {return Q.isEmpty();}
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/09-Queue-Deque/LeetCode/LeetCode-0225-2.cpp
/// Use a FIFO queue
class MyStack {
    queue<int> Q;

public:
    /** Initialize your data structure here. */
    MyStack() {}

    /** Push element x onto stack. */
    void push(int x) {
        int size = Q.size();
        Q.push(x);
        while (size-- > 0){
            Q.push(Q.front());
            Q.pop();
        } // end-for
    } //end-push

    /** Removes the element on top of the stack and returns that element. */
    int pop() {
        int item = Q.front();
        Q.pop();
        return item;
    } //end-pop

    /** Get the top element. */
    int top() {
		return Q.front();
    } //end-top

    /** Returns whether the stack is empty. */
    bool empty() {
        return Q.empty();
    } //end-empty
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */